smartpy/initializers.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import theano
+
+floatX = theano.config.floatX
+
+class WeightInitializer:
+    def __init__(self, random_seed=None):
+        self.rng = np.random.mtrand.RandomState(random_seed)
+
+    def __call__(self, dim):
```

Comment:
Instead of taking a shape in argument, what do you think about taking directly the `SharedVariable` object? That would reduce the code when calling the initer:
`weights_initializer(self.W)` 
instead of
`self.W.set_value(weights_initializer(self.W.get_value().shape))` 

The object is modified *in place* though, what do you think?